include/triton/Tools/Sys/GetEnv.hpp

```diff
@@ -32,7 +32,7 @@
 namespace mlir::triton {
 
 inline const std::set<std::string> ENV_VARS = {
-    "AMDGCN_ENABLE_DUMP",
```

Comment:
Why is this env var removed here? `amd/backend/compiler.py` is still using it.

Reply:
I believe this file only needs to list envvars that are used from C++

Reply:
Hmm, IIUC it is wrapped to the Python frontend here https://github.com/openai/triton/blob/85fc427dfc3e15bc2d26b8b395113fdc94f69831/python/src/ir.cc#L1618 and 
used to make sure the cache is invalidated here https://github.com/openai/triton/blob/85fc427dfc3e15bc2d26b8b395113fdc94f69831/python/triton/compiler/compiler.py#L239

Reply:
OK I didn't know about that, but still that only needs to / should apply to envvars that affect the compiled code.  If you read an envvar only from Python and it doesn't affect the generated code, it doesn't strictly need to be in this list.

Maybe it should be in the list, but we break that rule in like a dozen places.

Reply:
I actually think here we want to recompile when we turn that env var on, because it won't get printed otherwise ? :p

Reply:
Yeah that's fair. This dict should be renamed `CACHE_INVALIDATING_ENV_VARS`. I will submit a PR.

Reply:
Yeah, I was going by what @jlebar  pointed out. The triton::tools::getBoolEnv was the only place I saw that was checking the env was in the ENV_VARS table, but the `os.environ.get`'s used from Python aren't checking any such table.